### PR TITLE
[libs/ui] Add shared UiString component and associated context provider

### PR DESCRIPTION
**Pull request by kofi-q**
## Overview

_Easier by commit_

_Related spec [here](https://docs.google.com/document/d/1a2GRj1g4aOawEssPdTCgbv-qs28d-M_dyMEfzm4I4nM/edit?usp=sharing)._

Adding a `<UiString>` base component for all voter-facing strings that need to be translated and/or spoken, along with a `<UiStringsContextProvider>` to manage the underlying translation/audio data.

## Demo Video or Screenshot

https://github.com/votingworks/vxsuite/assets/264902/5a44baa9-38a8-470e-9c42-f0beabdf463d

## Testing Plan
- Unit tests for the base component and context provider
- Storybook demo for manual verification of functionality/interaction

## Checklist

- ~[ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.~

- [x] I have added a screenshot and/or video to this PR to demo the change
- ~[ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates~

### CODE REVIEW

File: libs/ui/src/ui_strings/language_context.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    i18next.addResourceBundle(currentLanguageCode, DEFAULT_I18NEXT_NAMESPACE, {
+      ...uiStringsData,
+    });
+
+    void i18next.changeLanguage(currentLanguageCode);
+  }, [currentLanguageCode, isUiStringsLoading, uiStringsData]);
+
+  if (!isI18nextReady || availableLanguagesQuery.isLoading) {
```
Comment (arsalansufi) [inline libs/ui/src/ui_strings/language_context.tsx:86]: You could alternatively use `!availableLanguagesQuery.isSuccess`, which will cover the initial loading state but will also make the `assertDefined` around the `availableLanguagesQuery.data` below unnecessary

File: libs/ui/src/ui_strings/language_context.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    // This state is too brief to warrant a loading screen which would only
+    // flash for an instant - going with an empty screen, since this only
+    // happens once on initial app render, before any content is rendered.
```
Comment (arsalansufi) [inline libs/ui/src/ui_strings/language_context.tsx:89]: Makes sense! Similar to what we do elsewhere in the UI

File: libs/ui/test/ui_strings/test_strings.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  planetName: (planetId: string) => (
+    <UiString uiStringKey="planetName" uiStringSubKey={planetId}>
+      {planetId}
+    </UiString>
+  ),
+} as const;
+
+// Using `testUiStrings.numPlanets()` directly in Storybook, doesn't work, for
+// some reason. This is a convenience wrapper to avoid the failure, which seems
+// related to rendering the count param object as a React component child.
```
Comment (arsalansufi) [inline libs/ui/test/ui_strings/test_strings.tsx:57]: Interesting 🤔 - will be curious to see if this error comes up in any other contexts

File: libs/ui/test/ui_strings/test_strings.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+export const TEST_UI_STRING_KEY = 'numPlanets';
+
+export const TEST_UI_STRING_TRANSLATIONS: UiStringsPackage = {
+  [LanguageCode.ENGLISH]: {
+    [`numPlanets_one`]: 'There is only <2>{{count}}</2> planet.',
+    [`numPlanets_other`]: 'There are <2>{{count}}</2> planets.',
+    planetName: {
+      planet1: 'Mercury',
+      planet3: 'Earth',
+      planet9: 'Pluto', // #NeverForget
```
Comment (arsalansufi) [inline libs/ui/test/ui_strings/test_strings.tsx:18]: Was going to comment on the boldness of your decision to still include Pluto after seeing it in the PR description 😆

File: libs/ui/src/ui_strings/ui_strings_context.test.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  const { getAudioContext, getLanguageContext, mockBackendApi, render } =
+    newTestContext({ noAudio: true });
+  mockBackendApi.getAvailableLanguages.mockResolvedValue([
+    LanguageCode.ENGLISH,
+  ]);
+  mockBackendApi.getUiStrings.mockResolvedValue(null);
+
+  render(<div>foo</div>);
+
+  expect(getLanguageContext).toBeDefined();
```
Comment (arsalansufi) [inline libs/ui/src/ui_strings/ui_strings_context.test.tsx:29]: I think you probably meant to add parentheses after `getLanguageContext` here (and wrap in a `waitFor` as needed)

**Review by arsalansufi — APPROVED** [review]
LGTM!

File: libs/ui/src/ui_strings/ui_strings_context.test.tsx (hunk 5 above)
Comment (kofi-q) [reply libs/ui/src/ui_strings/ui_strings_context.test.tsx:29]: oh, thanks for catching 😅

**Pull request merged**